### ISSUE TITLE
ArithmeticException: uint64 underflow pre-genesis on Spadina

### ISSUE THREAD

**Issue by benjaminion** [Consensys-Incorporated/teku#2854]
When running on Spadina pre-genesis (after the genesis state was generated, but before the genesis event), the following occurs 4 - 5 minutes after startup. It is repeatable.

I don't know why we are running forkchoice pre-genesis 🤔 

Version: `teku/v0.12.8-dev-b932f9fd/linux-x86_64/-ubuntu-openjdk64bitservervm-java-11`

```
2020-09-27 17:42:20.323+01:00 | nioEventLoopGroup-3-4 | ERROR | Subscribers | Error in callback:
java.lang.ArithmeticException: uint64 underflow
        at tech.pegasys.teku.infrastructure.unsigned.UInt64.minus(UInt64.java:160) ~[teku-infrastructure-unsigned-0.12.8-SNAPSHOT.jar:0.12.8-dev-b932f9fd]
        at tech.pegasys.teku.infrastructure.unsigned.UInt64.minus(UInt64.java:155) ~[teku-infrastructure-unsigned-0.12.8-SNAPSHOT.jar:0.12.8-dev-b932f9fd]
        at tech.pegasys.teku.core.ForkChoiceUtil.getCurrentSlot(ForkChoiceUtil.java:62) ~[teku-ethereum-core-0.12.8-SNAPSHOT.jar:0.12.8-dev-b932f9fd]
        at tech.pegasys.teku.core.ForkChoiceUtil.get_slots_since_genesis(ForkChoiceUtil.java:58) ~[teku-ethereum-core-0.12.8-SNAPSHOT.jar:0.12.8-dev-b932f9fd]
        at tech.pegasys.teku.core.ForkChoiceUtil.get_current_slot(ForkChoiceUtil.java:70) ~[teku-ethereum-core-0.12.8-SNAPSHOT.jar:0.12.8-dev-b932f9fd]
        at tech.pegasys.teku.core.ForkChoiceUtil.get_current_slot(ForkChoiceUtil.java:74) ~[teku-ethereum-core-0.12.8-SNAPSHOT.jar:0.12.8-dev-b932f9fd]
        at tech.pegasys.teku.storage.client.RecentChainData.getCurrentSlot(RecentChainData.java:296) ~[teku-storage-0.12.8-SNAPSHOT.jar:0.12.8-dev-b932f9fd]
        at tech.pegasys.teku.sync.singlepeer.SyncManager.peerStatusIsConsistentWithOurNode(SyncManager.java:272) ~[teku-sync-0.12.8-SNAPSHOT.jar:0.12.8-dev-b932f9fd]
        at tech.pegasys.teku.sync.singlepeer.SyncManager.isPeerSyncSuitable(SyncManager.java:266) ~[teku-sync-0.12.8-SNAPSHOT.jar:0.12.8-dev-b932f9fd]
        at tech.pegasys.teku.sync.singlepeer.SyncManager.onNewPeer(SyncManager.java:251) ~[teku-sync-0.12.8-SNAPSHOT.jar:0.12.8-dev-b932f9fd]
        at tech.pegasys.teku.networking.eth2.peers.Eth2PeerManager.lambda$onConnect$11(Eth2PeerManager.java:208) ~[teku-networking-eth2-0.12.8-SNAPSHOT.jar:0.12.8-dev-b932f9fd]
        at tech.pegasys.teku.infrastructure.subscribers.Subscribers.lambda$forEach$0(Subscribers.java:98) ~[teku-infrastructure-subscribers-0.12.8-SNAPSHOT.jar:0.12.8-dev-b932f9fd]
        at java.util.concurrent.ConcurrentHashMap$ValuesView.forEach(ConcurrentHashMap.java:4772) ~[?:?]
        at tech.pegasys.teku.infrastructure.subscribers.Subscribers.forEach(Subscribers.java:95) ~[teku-infrastructure-subscribers-0.12.8-SNAPSHOT.jar:0.12.8-dev-b932f9fd]
        at tech.pegasys.teku.networking.eth2.peers.Eth2PeerManager.lambda$onConnect$12(Eth2PeerManager.java:208) ~[teku-networking-eth2-0.12.8-SNAPSHOT.jar:0.12.8-dev-b932f9fd]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$finish$18(SafeFuture.java:258) ~[teku-infrastructure-async-0.12.8-SNAPSHOT.jar:0.12.8-dev-b932f9fd]
        at java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:930) ~[?:?]
        at java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:907) ~[?:?]
        at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
        at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073) ~[?:?]
        at tech.pegasys.teku.networking.eth2.peers.Eth2Peer.lambda$updateStatus$0(Eth2Peer.java:97) ~[teku-networking-eth2-0.12.8-SNAPSHOT.jar:0.12.8-dev-b932f9fd]
        at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$finish$18(SafeFuture.java:258) ~[teku-infrastructure-async-0.12.8-SNAPSHOT.jar:0.12.8-dev-b932f9fd]
etc...
```

**Comment by cemozerr** [comment]
I noticed that I don't see this error when I delete the /data/db directory.

**Comment by hwwhww** [comment]
I had the same issue yesterday, and then #2863 fixed my teku node. 👍

**Comment by cemozerr** [comment]
Closing this as it seems to be solved with #2863.
